GetIntoTeachingApi/Models/Location.cs

```diff
@@ -9,8 +9,13 @@ namespace GetIntoTeachingApi.Models
 {
     public class Location
     {
+        public static readonly Regex AreaOrFullPostcodeRegex = new Regex(
+            $"{AreaPostcodePattern}|{FullPostcodePattern}", RegexOptions.IgnoreCase);
         public static readonly Regex PostcodeRegex = new Regex(
-            @"^([A-Z][A-HJ-Y]?\d[A-Z\d]? ?\d[A-Z]{2}|GIR ?0A{2})$", RegexOptions.IgnoreCase);
+            FullPostcodePattern, RegexOptions.IgnoreCase);
+
+        private const string AreaPostcodePattern = @"^([A-Z][A-HJ-Y]?\d[A-Z\d]?)$";

```

Comment:
Can we rename this to `OutwardPostcodePattern`? a postcode "area" refers to the first one or two characters, not including the digits so it's a little misleading. 

It may also be worth having your private constants as `OutwardPostcodePattern` and `InwardPostcodePattern` then combining them in the public constants `OutwardOrFullPostcodeRegex` and `PostcodeRegex` as a way of DRYing up the outward pattern (or interpolate the outward pattern into the full pattern const and leave it as you have it).